staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -273,12 +273,16 @@ func (j *JSONPath) evalArray(input []reflect.Value, node *ArrayNode) ([]reflect.
 			return result, nil
 		}
 
-		if !params[2].Known {
-			value = value.Slice(params[0].Value, params[1].Value)
-		} else {
-			value = value.Slice3(params[0].Value, params[1].Value, params[2].Value)
+		value = value.Slice(params[0].Value, params[1].Value)
+
+		step := 1
+		if params[2].Known {
+			if params[2].Value <= 0 {
+				return input, fmt.Errorf("step must be >= 0")
```

Comment:
A bit curious: why is step of zero allowed ?

Reply:
it is not allowed. a step size of 0 means that no progress would be made between the start and end

Reply:
oh, I see... the message is incorrect

Reply:
Should I send out PR to correct the message ?

Reply:
sure, thanks

Reply:
Created #73179